drivers/QSPI.cpp

```diff
@@ -0,0 +1,294 @@
+    
+    if(_initialized) {
+        if( (tx_length != NULL) && (tx_buffer != NULL) ) {
+            if(*tx_length != 0) {
+                lock();
+                if(true == _acquire()) {
+                    qspi_command_t *qspi_cmd = _build_qspi_command(-1, address, -1);
+                    if(QSPI_STATUS_OK == qspi_write(&_qspi, qspi_cmd, tx_buffer, tx_length)) {
+                        ret_status = QSPI_STATUS_OK; 
+                    }
+                }
+                unlock();
+            }
+        } else {
+            ret_status = QSPI_STATUS_INVALID_PARAMETER; 
+        }
+    } //if(_initialized)
+    
+    return ret_status;
+}
+
+qspi_status_t QSPI::read(unsigned int instruction, unsigned int address, unsigned int alt, char *rx_buffer, size_t *rx_length) 
+{
+    qspi_status_t ret_status = QSPI_STATUS_ERROR; 
+    
+    if(_initialized) {
+        if( (rx_length != NULL) && (rx_buffer != NULL) ) {
+            if(*rx_length != 0) {
+                lock();
+                if( true == _acquire()) {
+                    qspi_command_t *qspi_cmd = _build_qspi_command(instruction, address, alt);
+                    if(QSPI_STATUS_OK == qspi_read(&_qspi, qspi_cmd, rx_buffer, rx_length)) {
+                        ret_status = QSPI_STATUS_OK; 
+                    }
+                }
+                unlock();
+            }
+        } else {
+            ret_status = QSPI_STATUS_INVALID_PARAMETER; 
+        }
+    } //if(_initialized)
+    
+    return ret_status;
+}
+
+qspi_status_t QSPI::write(unsigned int instruction, unsigned int address, unsigned int alt, const char *tx_buffer, size_t *tx_length) 
+{
+    qspi_status_t ret_status = QSPI_STATUS_ERROR; 
+    
+    if(_initialized) {
+        if( (tx_length != NULL) && (tx_buffer != NULL) ) {
+            if(*tx_length != 0) {
+                lock();
+                if(true == _acquire()) {
+                    qspi_command_t *qspi_cmd = _build_qspi_command(instruction, address, alt);
+                    if(QSPI_STATUS_OK == qspi_write(&_qspi, qspi_cmd, tx_buffer, tx_length)) {
+                        ret_status = QSPI_STATUS_OK; 
+                    }
+                }
+                unlock();
+            }
+        } else {
+            ret_status = QSPI_STATUS_INVALID_PARAMETER; 
+        }
+    } //if(_initialized)
+    
+    return ret_status;
+}
+
+qspi_status_t QSPI::command_transfer(unsigned int instruction, const char *tx_buffer, size_t tx_length, const char *rx_buffer, size_t rx_length) 
+{
+    qspi_status_t ret_status = QSPI_STATUS_ERROR; 
+    
+    if(_initialized) {
+        lock();
+        if(true == _acquire()) {
+            qspi_command_t *qspi_cmd = _build_qspi_command(instruction, -1, -1); //We just need the command
+            if(QSPI_STATUS_OK == qspi_command_transfer(&_qspi, qspi_cmd, (const void *)tx_buffer, tx_length, (void *)rx_buffer, rx_length)) {
+                //We got error status, return 0
```

Comment:
error status? we got OK? shall we just remove  this comment

Reply:
Oh yeah, its from older code structure, lets remove it.